### PR TITLE
Implement async workers

### PR DESCRIPTION
**Pull request by jsvana**
Makes `bounce` function as a basic IRC proxy that responds to pings in the background.

### CODE REVIEW

File: src/irc/types.rs [hunk 1]
```diff
@@ -75,10 +75,21 @@ impl FromStr for Prefix {
 #[derive(Debug)]
 pub struct Message {
     prefix: Option<Prefix>,
+    // TODO(jsvana): Maybe make this an enum?
     command: String,
     params: Option<Vec<String>>,
```
Comment (belak) [inline src/irc/types.rs:80]: Could this just be a Vec<String>?

Comment (jsvana) [reply src/irc/types.rs:80]: Sorry I missed this! Fixed in ae17cbf.

Comment (belak) [reply src/irc/types.rs:80]: Github queued up the comment and I forgot to accidentally submit it

File: src/main.rs [hunk 2]
```diff
@@ -1,56 +1,107 @@
...
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let mut listener = TcpListener::bind("127.0.0.1:49654").await?;
+
+    println!("Listening on 49654");
+    loop {
+        let (socket, _) = listener.accept().await?;
+        println!("Got new connection");
+        start_workers(socket).await?;
```
Comment (guswynn) [inline src/main.rs]: this will block the ability to accept new listeners

File: src/main.rs [hunk 3]
```diff
@@ -1,44 +1,25 @@
...
-struct IrcServer {
-    name: String,
-    host_port: SocketAddr,
-    // password: String,
-}
-
 async fn read_worker<T, U>(
-    _server_reader: tokio::io::ReadHalf<T>,
-    _user_writer: tokio::io::WriteHalf<U>,
+    mut server_reader: tokio::io::ReadHalf<T>,
```
Comment (guswynn) [inline src/main.rs]: why are these mutable?

File: src/config.rs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pub core: Core,
+    pub networks: Vec<Network>,
+}
+
+impl Config {
+    pub fn from_file(filename: &str) -> Result<Self> {
+        toml::from_str(&std::fs::read_to_string(filename)?)
```
Comment (guswynn) [inline src/config.rs:59]: serde toml seems dankkk

**Review by guswynn — COMMENTED** [review]
some rando comments

File: config.toml.example [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+[[networks]]
+name = "some_network"
+nick = "coolguy42"
+alt_nick = "coolguy42_"
+ident = "coolguy42"
+realname = "Cool Guy"
+
+  [networks.server]
+  hostname = "irc.freenode.net"
```
Comment (guswynn) [inline config.toml.example:13]: test comment so I can figure out github

**Review by guswynn — COMMENTED** [review]
test

**Review by guswynn — COMMENTED** [review]
another

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,4 @@
 /target
 **/*.rs.bk
+
```
Comment (guswynn) [inline .gitignore:3]: test2

**Pull request merged**